### PR TITLE
Remove hardcode reuqest of friends & Load author profile page and send friend request & Search bar works!!

### PR DESCRIPTION
**Pull request by XiaoleZ**
Remove hardcode reuqest of friends & Load author profile page and send friend request.

`http://<service>/author/<author_id>/profile` can view the author's profile
If the author from the url is the user, then the user is able to edit the profile.
If the author from the url is not the user, then the user is able to Add friend/Send request.

Screen Shot 2020-03-11 at 10 03 30 PM

The search bar can search by `firstName`, `lastName` and `displayName`
Screen Shot 2020-03-12 at 2 12 34 AM

### CODE REVIEW

File: socialdistribution/socialdistribution/urls.py [hunk 1]
```diff
@@ -31,7 +31,9 @@
     path('new_post/', include('profiles.urls')),
     path('editprofile/', profiles_views.edit_profile, name='editprofile'),
     path('viewprofile/', profiles_views.view_profile, name='viewprofile'),
-
+    path(
+        'author/<uuid:author_id>/profile', profiles_views.view_author_profile, name='view_author_profile'
```
Comment (um4r12) [inline socialdistribution/socialdistribution/urls.py:35]: What is the difference between `author/<uuid:author_id>/profile` and `/stream/author/<uuid:author_id>/profile`?  Why were both added?

Comment (XiaoleZ) [reply socialdistribution/socialdistribution/urls.py:35]: Just for testing. removed.

File: socialdistribution/profiles/views.py [hunk 2]
```diff
@@ -99,6 +99,25 @@ def edit_profile(request):
...
+    #The user who login in/use the application
+    # TODO: add cookie or token to store the user
+    user_author = request.user
+    
+    author = Author.objects.get(id=author_id)
+    template = 'profiles/profiles_view.html'
+    # form = ProfileForm(instance=author)
+    status = True
+
+    if author == user_author:
```
Comment (um4r12) [inline socialdistribution/profiles/views.py:113]: You could just redirect to `/viewprofile` if `author == user_author` instead duplicating the template code using the status flag. Or scrap the `viewprofile` and just use this.

Comment (XiaoleZ) [reply socialdistribution/profiles/views.py:113]: I think, even if the user itself, it can be see in such url.
Therefore, I would keep it as now.

Comment (um4r12) [reply socialdistribution/profiles/views.py:113]: Sure, but it just causes inconsistency. There are two urls that handle the same functionality just differently depending on status.

File: socialdistribution/profiles/views.py [hunk 3]
```diff
@@ -1,8 +1,9 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render, redirect, render_to_response
```
Comment (um4r12) [inline socialdistribution/profiles/views.py]: `render_to_response` is deprecated and removed in django 3+ and does not run. Remove import also not used anywhere.

**Pull request merged**